crates/wasmtime/src/runtime/store.rs

```diff
@@ -2264,6 +2424,41 @@ at https://bytecodealliance.org/security.
 
         Ok(id)
     }
+
+    /// Set a pending exception. The `exnref` is taken and held on
+    /// this store to be fetched later by an unwind. This method does
+    /// *not* set up an unwind request on the TLS call state; that
+    /// must be done separately.
+    #[cfg(feature = "gc")]
+    pub(crate) fn set_pending_exception(&mut self, exnref: VMExnRef) {
+        debug_assert!(self.pending_exception.is_none());
+        self.pending_exception = Some(exnref);
+    }
+
+    /// Take a pending exception, if any.
+    #[cfg(feature = "gc")]
+    pub(crate) fn take_pending_exception(&mut self) -> Option<VMExnRef> {
+        self.pending_exception.take()
+    }
+
+    #[cfg(feature = "gc")]
+    fn take_pending_exception_rooted(&mut self) -> Option<Rooted<ExnRef>> {
+        self.take_pending_exception().map(|vmexnref| {
```

Comment:
You can use `?` after `self.take_pending_exception()` to avoid indentation if you'd like